### DISCUSSION TITLE
Release build/Production settings

### DISCUSSION THREAD

**Discussion by pme442** [AdamEssenmacher/MemoryToolkit.Maui#4] [Q&A] [answered]
First of all, Adam, thank you for this.  It has really helped me to understand and identify the leaks in my app.
I know that you said, "you might not want to use leak detection in release builds".  
Just to clarify, does that mean release builds should set AutoDisconnectBehavior.Cascade="True", and GCMonitorBehavior.Cascade="False"? (Since the forced GC calls are only being done in the GCCollectionMonitor.)

**Comment by AdamEssenmacher** [comment, marked as answer]
Yes, your understanding is correct.

The only reason you _might_ want to use `GCMonitorBehavior.Cascade="True"` in a release build is to detect leaks in production. Leak detection causes some overhead (both in walking the visual tree and in running the GC), so it's best to use it only during development.

`AutoDisconnectBehavior`'s purpose is to fix memory leaks where they occur. So if you're relying on it to prevent a leak, it has to be set to 'True' in production.
